test/controllers/user_blocks_controller_test.rb

```diff
@@ -483,13 +483,44 @@ def test_update_expired
     assert_equal "Original Reason", block.reason
 
     session_for(creator_user)
+    check_block_updates(block)
+  end
+
+  ##
+  # test the update action on revoked blocks
+  def test_update_revoked
+    creator_user = create(:moderator_user)
+    revoker_user = create(:moderator_user)
+    other_moderator_user = create(:moderator_user)
+    block = create(:user_block, :revoked, :creator => creator_user, :revoker => revoker_user, :reason => "Original Reason")
+
+    session_for(other_moderator_user)
+    put user_block_path(block,
+                        :user_block_period => "0",
+                        :user_block => { :needs_view => false, :reason => "Updated Reason" })
+    assert_redirected_to edit_user_block_path(block)
+    assert_equal "Only the moderators who created or revoked this block can edit it.", flash[:error]
+    block.reload
+    assert_not_predicate block, :active?
+    assert_equal "Original Reason", block.reason
+
+    session_for(creator_user)
+    check_block_updates(block)
+
+    session_for(revoker_user)
+    check_block_updates(block)
+  end
+
+  private
+
+  def check_block_updates(block)
```

Comment:
Can we move this to the end with the other private methods rather than switching back and forth between public and private?

Reply:
Do you want to move only `check_block_updates` or every other such method too?

Reply:
Oh I hadn't realised you had done the same thing in that other PR that I merged yesterday... Personally I would prefer to move that as well.

Reply:
I moved the methods here. But I didn't do a similar private section just yesterday, there should be more of them elsewhere.